lib/ansible/modules/cloud/docker/docker_container.py

```diff
@@ -2265,8 +2273,14 @@ def has_different_resource_limits(self):
             memory=host_config.get('Memory'),
             memory_reservation=host_config.get('MemoryReservation'),
             memory_swap=host_config.get('MemorySwap'),
+            restart_policy=host_config.get('RestartPolicy')
         )
 
+        # Options which don't make sense without their accompanying option
+        if self.parameters.restart_policy:
+            restart_policy = host_config.get('RestartPolicy', dict())
```

Comment:
Here you need to reuse it.

Reply:
Also addressed with 808987abac2e1e95f21db51def333bbb18ffd68e.